### PR TITLE
view toolbar for avg waveforms view

### PR DESCRIPTION
**Pull request by magland**
Porting view toolbar functionality from sortingview-gui
Control amplitude scaling and waveform layout mode

image

### CODE REVIEW

File: src/views/AverageWaveforms/AverageWaveformsView.tsx [hunk 1]
```diff
@@ -19,35 +23,103 @@ const AverageWaveformsView: FunctionComponent<Props> = ({data, width, height}) =
...
+                title: 'Scale amplitude down [down arrow]',
+                icon: <FaArrowDown />,
+                keyCode: 40
+            },
+            {
+                type: 'text',
+                title: 'Zoom level',
+                content: ampScaleFactor,
+                contentSigFigs: 2
+            },
```
Comment (jsoules) [inline src/views/AverageWaveforms/AverageWaveformsView.tsx:87]: I'd suggest incorporating some of the changes I'd made to make the toolbar into a shared component--see the `14-toolbar-synthesis` branch of figurl, specifically [the AmplitudeScaleToolbarEntries](https://github.com/magland/figurl/blob/cba9c168ea061fa7b56a525229a40cdc2eda4bc1/src/plugins/sortingview/gui/extensions/common/sharedToolbarSets/AmplitudeScaleToolbarEntries.tsx) and [AverageWaveformsView that consumes it](https://github.com/magland/figurl/blob/cba9c168ea061fa7b56a525229a40cdc2eda4bc1/src/plugins/sortingview/gui/extensions/averagewaveforms/AverageWaveformsView/AverageWaveformsView.tsx#L56-L75).

This should make it easier to have a consistent interface across places we want to implement the same action (scaling in this case, but I also made one for time widget operations).

File: src/views/common/ViewToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
+import { Checkbox, IconButton } from '@material-ui/core';
```
Comment (jsoules) [inline src/views/common/ViewToolbar.tsx:1]: Again, I'd done some reworking of this guy in the `14-toolbar-synthesis` branch of Figurl, see [updated ViewToolbar.tsx](https://github.com/magland/figurl/blob/14-toolbar-synthesis/src/plugins/sortingview/gui/extensions/common/ViewToolbar.tsx) (that among other things adds support for sliders) and [Toolbars.ts](https://github.com/magland/figurl/blob/14-toolbar-synthesis/src/plugins/sortingview/gui/extensions/common/Toolbars.ts) that defines the relevant types.

**Review by jsoules — COMMENTED** [review]
Generally looks good, but I think this was copied from the main branch of figurl--I had quite a few changes on the `14-toolbar-synthesis` branch that I think would be improvements & would like to include. LMK if you'd like to discuss.

**Comment by magland** [comment]
I have moved the newest code over from:

https://github.com/magland/figurl/blob/14-toolbar-synthesis/src/plugins/sortingview/gui/extensions/common/ViewToolbar.tsx

https://github.com/magland/figurl/blob/14-toolbar-synthesis/src/plugins/sortingview/gui/extensions/common/Toolbars.ts

https://github.com/magland/figurl/blob/14-toolbar-synthesis/src/plugins/sortingview/gui/extensions/common/sharedToolbarSets/AmplitudeScaleToolbarEntries.tsx

**Review by jsoules — COMMENTED** [review]
This looks good to me.

**Pull request merged**